animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -225,17 +247,20 @@ cdef class Mesh:
                 self.getMeshProperties(), source.getMeshProperties()):
             for name in meshProperty.keys():
                 if name in sourceMeshProperty:
-                    meshProperty[name].extend(sourceMeshProperty[name])
+                    meshProperty[name].extend(sourceMeshProperty[name].data)
                 else:
-                    extension = meshPropertyType(length = source.polygons.indices.length)
+                    attribute = meshProperty[name]
+                    extension = attribute.getDataType()(length = source.polygons.indices.length)
                     extension.fill(0)
                     meshProperty[name].extend(extension)
 
             for name in sourceMeshProperty.keys():
                 if name not in meshProperty:
-                    extension = meshPropertyType(length = self.polygons.indices.length)
+                    attribute = sourceMeshProperty[name]
+                    extension = attribute.getDataType()(length = self.polygons.indices.length)
                     extension.fill(0)
-                    meshProperty[name] = extension + sourceMeshProperty[name]
+                    data = extension + attribute.data
+                    meshProperty[name] = Attribute(name, attribute.type, attribute.domain, attribute.dataType, data)
```

Comment:
Most of this should be be moved to the attribute class. Instead of having a `extend` method in the attribute class, make it `appendAttribute` or something like that. The same for for other attribute methods.

Reply:
Added `appendAttribute` method.